docs-src/typescript/how-to-use-worker-versioning-in-typescript.md

```diff
@@ -0,0 +1,83 @@
+---
+id: how-to-use-worker-versioning-in-typescript
+title: How to use Worker Versioning in TypeScript
+sidebar_label: Worker Versioning
+description: Version your TypeScript Workers by using build ID-based versioning
+tags:
+  - typescript
+  - how-to
+---
+
+To use Worker Versioning in TypeScript, you need to do the following:
+
+1. Determine and assign a Build ID to your built Worker code, and opt in to versioning.
+2. Tell the Task Queue your Worker is listening on about that Build ID, and whether its compatible with an existing Build ID.
+
+### Assign a Build ID to your Worker
+
+Let's say you've chosen `deadbeef` as your Build ID, which might be a short git commit hash (a reasonable choice as Build ID).
+To assign it in your Worker code, assign the following Worker Options:
+
+```typescript
+// ...
+const worker = await Worker.create({
+  taskQueue: 'your_task_queue_name',
```

Comment:
```suggestion
  taskQueue: 'your-task-queue-name',
```

underscores feels weird in TS to me, i think all our samples use `-`

Reply:
Gonna keep this to be consistent with the other very similar guides